### PR TITLE
`/submit-network` memory optimization (part 3)

### PR DESCRIPTION
**Pull request by ricksu978**
Changes
- use pipe for unzip | hash | parse_weight

Uploading 256x20 Network (Gzipped 88.9MB, Unzipped 260MB)
```
submit-network begins
        rss       231.29 MB
        heapTotal 31.33 MB
        heapUsed  17.74 MB
        external  195.07 MB
unzip completed
        rss       269.43 MB
        heapTotal 32.33 MB
        heapUsed  15.91 MB
        external  295.15 MB
::1 (undefined)  uploaded network 00335835cc85548b1b22d79e7df420ab38b9e2b7117a0e18ae9c5cb7a81559fb (175000)
Network weights  (256 x 20) 00335835cc85548b1b22d79e7df420ab38b9e2b7117a0e18ae9c5cb7a81559fb already exists.
```

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -315,7 +321,7 @@ function how_many_games_to_queue(max_games, w_obs, l_obs, pessimistic_rate) {
 
 app.enable('trust proxy');
 
-app.use(bodyParser.urlencoded({extended: true}));
+//app.use(bodyParser.urlencoded({extended: true}));
```
Comment (roy7) [inline server.js]: Is bodyParser not actually needed? I added it on long ago when first starting this project, just based on some examples I as starting out from. (This was my first Node project from scratch and first use of Express.) We don't lose access to req.body in various locations with it gone?

**Comment by rchs819** [comment]
`bodyParse` is still needed.

I was experimenting something else.
There are still rooms for memory optimization using `busbody`.

@roy7 do you know much RAM we have on the server, and how much RAM is used by the nodejs constantly?

**Comment by roy7** [comment]
A quick 'top' snapshot.

```
Tasks: 115 total,   1 running, 114 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.5 us,  0.2 sy,  0.0 ni, 99.2 id,  0.2 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem :  2048072 total,   126072 free,  1094840 used,   827160 buff/cache
KiB Swap:  2812924 total,  2693880 free,   119044 used.   757920 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
31051 mongodb   20   0 2020592 833248   3756 S   1.0 40.7 126:09.15 mongod
 1493 jroy      20   0 1281564  97504   8132 S   0.7  4.8  24:00.70 node
30919 nginx     20   0   47832  10392   6376 S   0.7  0.5  80:17.01 nginx
14537 jroy      20   0   44520   3544   2980 R   0.3  0.2   0:00.01 top
```

**Comment by roy7** [comment]
Sounds like busboy might let us stream the uploading files directly to disk, then we'd process them via streams after upload is complete, vs the whole upload being stored in the req.files in ram?

**Comment by rchs819** [comment]
Yes, correct. pipe, pipe and pipe.

A quick peeps on my experiment using `busboy`

Uploading 256x20 Network (Gzipped 88.9MB, Unzipped 260MB)
```
submit start
        rss       36.95 MB
        heapTotal 15.83 MB
        heapUsed  12.7 MB
        external  17.23 MB
submit ends
        rss       53.43 MB
        heapTotal 19.33 MB
        heapUsed  13.93 MB
        external  31.01 MB
```

**Pull request merged**